### DISCUSSION TITLE
Negative scores with BASE and DCQ methods on deconvolute_mouse function?

### DISCUSSION THREAD

**Discussion by erzakiev** [omnideconv/immunedeconv#136] [General]
Hello, so I was wondering if it's normal that both the DCQ and BASE methods produce abundance matrices that contain negative values?

> knitr::kable(immunedeconv_results_dcq, digits = 2)

|cell_type                       | 25_tpm_ctrl| 41_tpm_ctrl| 42_tpm_ctrl| 44_tpm| 45_tpm| 46_tpm|
|:-------------------------------|-----------:|-----------:|-----------:|------:|------:|------:|
|B cell                          |        0.02|       -0.05|       -0.31|   0.34|  -0.03|   0.04|
|B cell naive                    |        0.01|       -0.04|       -0.24|   0.26|  -0.03|   0.03|
|Myeloid dendritic cell          |        0.03|       -0.07|       -0.47|   0.51|  -0.05|   0.06|
|Plasmacytoid dendritic cell     |        0.00|        0.01|        0.05|  -0.05|   0.01|  -0.01|
|Granulocyte-monocyte progenitor |        0.00|        0.00|       -0.01|   0.01|   0.00|   0.00|
|Macrophage                      |       -0.09|        0.21|        1.43|  -1.54|   0.16|  -0.18|
|Stem cell                       |        0.02|       -0.05|       -0.33|   0.35|  -0.04|   0.04|
|Monocyte                        |        0.01|       -0.02|       -0.10|   0.11|  -0.01|   0.01|
|NK cell                         |       -0.01|        0.02|        0.11|  -0.12|   0.01|  -0.01|
|T cell NK                       |        0.00|        0.00|        0.03|  -0.03|   0.00|   0.00|
|T cell immature                 |       -0.79|        1.90|       12.80| -13.75|   1.40|  -1.56|
|T cell CD4+ naive               |        0.00|        0.00|        0.03|  -0.03|   0.00|   0.00|
|T cell regulatory (Tregs)       |        0.00|        0.00|        0.01|  -0.01|   0.00|   0.00|
|T cell CD4+ memory              |        0.00|        0.00|        0.01|  -0.01|   0.00|   0.00|
|T cell CD8+ effector memory     |        0.00|        0.01|        0.05|  -0.05|   0.01|  -0.01|
|T cell CD8+ memory              |        0.00|        0.00|        0.02|  -0.03|   0.00|   0.00|
|T cell CD8+ naive               |        0.00|        0.01|        0.04|  -0.04|   0.00|   0.00|
|T cell gamma delta              |        0.02|       -0.05|       -0.32|   0.34|  -0.03|   0.04|
> knitr::kable(immunedeconv_results_base, digits = 2)

|cell_type                       | 25_tpm_ctrl| 41_tpm_ctrl| 42_tpm_ctrl| 44_tpm| 45_tpm| 46_tpm|
|:-------------------------------|-----------:|-----------:|-----------:|------:|------:|------:|
|B cell                          |        0.28|       -0.10|        0.62|  -0.23|  -0.33|  -0.53|
|B cell naive                    |       -0.64|        1.13|       -0.65|  -0.49|   0.10|   0.15|
|stroma score                    |       -1.35|        0.11|       -0.66|   0.76|   1.03|  -0.17|
|Myeloid dendritic cell          |        1.53|       -0.03|       -1.28|   0.44|  -0.89|   0.22|
|Granulocyte-monocyte progenitor |       -1.01|       -0.60|       -0.96|   1.54|   0.20|   0.49|
|Macrophage                      |       -0.92|        0.18|       -1.05|   1.09|   0.14|   0.87|
|Stem cell                       |        0.67|        0.91|       -0.69|  -1.27|  -0.27|  -0.29|
|NK cell                         |       -0.35|       -1.09|       -0.66|   0.58|   1.11|   0.59|
|T cell immature                 |       -0.58|        1.08|       -0.56|  -0.69|   0.13|  -0.01|
|T cell CD4+ naive               |        1.58|       -1.18|        0.25|   0.31|  -0.02|  -0.88|
|T cell CD4+ memory              |       -0.01|       -0.96|       -0.67|   0.20|   1.13|   0.46|
|T cell CD8+ effector memory     |        0.06|       -0.94|       -0.94|   0.74|   0.90|   0.42|
|T cell CD8+ memory              |        0.46|       -1.25|       -1.16|   0.62|   1.06|   0.35|
|T cell CD8+ naive               |        1.54|       -1.16|        0.29|   0.43|  -0.02|  -0.88|
|T cell gamma delta              |       -0.17|       -1.85|       -0.05|   0.90|   0.70|   0.47|

The input matrix that i use can be found here: 
[data4immunedeconv.txt](https://github.com/omnideconv/immunedeconv/files/10580990/data4immunedeconv.txt)

**Comment by LorenzoMerotto** [comment]
It it normal for these methods to produce negative results since: 
- DCQ applies a regression method which is not constrained
- BASE computes a score similar to gene set enrichment for each cell type 

So these results should be interpreted just as scores, and the negative values carry information as well.
